// @ethereum-attestation-service/eas-contracts/dist/typechain-types/factories/contracts/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaRegistry__factory = exports.ISchemaRegistry__factory = exports.IEAS__factory = exports.EAS__factory = exports.resolver = exports.eip712 = void 0;
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
exports.eip712 = __importStar(require("./eip712"));
exports.resolver = __importStar(require("./resolver"));
var EAS__factory_1 = require("./EAS__factory");
Object.defineProperty(exports, "EAS__factory", { enumerable: true, get: function () { return EAS__factory_1.EAS__factory; } });
var IEAS__factory_1 = require("./IEAS__factory");
Object.defineProperty(exports, "IEAS__factory", { enumerable: true, get: function () { return IEAS__factory_1.IEAS__factory; } });
var ISchemaRegistry__factory_1 = require("./ISchemaRegistry__factory");
Object.defineProperty(exports, "ISchemaRegistry__factory", { enumerable: true, get: function () { return ISchemaRegistry__factory_1.ISchemaRegistry__factory; } });
var SchemaRegistry__factory_1 = require("./SchemaRegistry__factory");
Object.defineProperty(exports, "SchemaRegistry__factory", { enumerable: true, get: function () { return SchemaRegistry__factory_1.SchemaRegistry__factory; } });
