// @ethereum-attestation-service/eas-contracts/dist/typechain-types/factories/contracts/ISchemaRegistry__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ISchemaRegistry__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "uid",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "address",
                name: "registerer",
                type: "address",
            },
        ],
        name: "Registered",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "uid",
                type: "bytes32",
            },
        ],
        name: "getSchema",
        outputs: [
            {
                components: [
                    {
                        internalType: "bytes32",
                        name: "uid",
                        type: "bytes32",
                    },
                    {
                        internalType: "contract ISchemaResolver",
                        name: "resolver",
                        type: "address",
                    },
                    {
                        internalType: "bool",
                        name: "revocable",
                        type: "bool",
                    },
                    {
                        internalType: "string",
                        name: "schema",
                        type: "string",
                    },
                ],
                internalType: "struct SchemaRecord",
                name: "",
                type: "tuple",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "string",
                name: "schema",
                type: "string",
            },
            {
                internalType: "contract ISchemaResolver",
                name: "resolver",
                type: "address",
            },
            {
                internalType: "bool",
                name: "revocable",
                type: "bool",
            },
        ],
        name: "register",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ISchemaRegistry__factory {
    static abi = _abi;
    static createInterface() {
        return new ethers_1.utils.Interface(_abi);
    }
    static connect(address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    }
}
exports.ISchemaRegistry__factory = ISchemaRegistry__factory;
